pkg/3scale/amp/component/system.go

```diff
@@ -772,6 +812,26 @@ func (system *System) SidekiqPodVolumes() []v1.Volume {
 	}
 
 	res = append(res, systemConfigVolume)
+
+	if system.Options.S3FileStorageOptions != nil && system.Options.S3FileStorageOptions.S3STSEnabled {
+		s3CredsProjectedVolume := v1.Volume{
+			Name: S3StsCredentialsSecretName,
+			VolumeSource: v1.VolumeSource{
+				Projected: &v1.ProjectedVolumeSource{
+					Sources: []v1.VolumeProjection{
+						v1.VolumeProjection{
+							ServiceAccountToken: &v1.ServiceAccountTokenProjection{
+								Audience:          "openshift",
+								ExpirationSeconds: &[]int64{3600}[0],
+								Path:              TokenName,
```

Comment:
Hardcoded? It should be read from the secret with 

```go
tokenFile := filepath.Base(secretFielfValue)
```

Reply:
done